.github/workflows/update-coverage-badge.yml

```diff
@@ -0,0 +1,69 @@
+name: Run Tests
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  update_coverage_badge:
+    name: Update Coverage Badge
+    runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: postgis/postgis:14-3.2-alpine
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_DB: tekdb
+        ports:
+          - 5432:5432
+    env:
+      SQL_ENGINE: django.contrib.gis.db.backends.postgis
+      SQL_DATABASE: tekdb
+      SQL_USER: postgres
+      SQL_PASSWORD: postgres
+      SQL_HOST: localhost
+      SQL_PORT: 5432
+      DEBUG: 1
+      SECRET_KEY: ${{ secrets.POSTGRES_SECRET_KEY }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v5
+
+      - name: Set up Python
+        uses: actions/setup-python@v6
+        with:
+          python-version: '3.10'
+
+      - name: Install system dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install gdal-bin -y
+
+      - name: Install Python dependencies
+        run: |
+          python -m pip install -r TEKDB/requirements.txt
+
+      - name: Run Coverage Report
+        run: |
+          cd TEKDB && coverage run manage.py test TEKDB.tests
+          coverage json
+      
+      - name: Access Coverage Stat
```

Comment:
got a lot of inspiration on how to make and update the coverage badge from this [article](https://nedbatchelder.com/blog/202209/making_a_coverage_badge.html)